### DISCUSSION TITLE
Usage in development

### DISCUSSION THREAD

**Discussion by JonCB** [featurehub-io/featurehub#593] [Q&A] [answered]
I want to get an idea of how people are using FeatureHub while developing on their local machine.

The first option I see is installing a local copy for dev purposes. Quick and simple, my only concern would be how much it uses in the way of machine resources .

The second option I see is just reusing the same server as every other environment. Does it slow things down to be connecting to the cloud to run things on your local machine? Does having 10 or so dev environments along with all the rest of the environments cause clutter

The third option would be to have a central server that is specifically for dev. Would still have the cloud latency/external connection issues but at least you wouldn't be cluttering the main server with additional environments.

Am I missing a strategy? What would be the generally recommended strategy?

Regards,
Jon

**Comment by rvowles** [comment, marked as answer]
The second is the way we tried to make it easy for people to use, when you end up using a server or environment that isn't the same as the one that is "real" you tend to end up missing things. The ability to hide other environments and have your own "personal" view of environments was inspired by this. 

The separation of the "repository" from how you connect to the repository however was also key to the idea that you could actually fill the repository with your own settings and configuration if you wished - say from a local file, and runtime determines where you actually get the features from. 

Thats how I have seen it, and how we have used it, but interested in any other POVs.
